src/submission/validation-clinical/validation.ts

```diff
@@ -29,11 +36,14 @@ import {
 import { concat } from 'lodash';
 import { ClinicalSubmissionRecordsOperations } from './utils';
 import { mergeRecordsMapIntoDonor } from '../submission-to-clinical/merge-submission';
+import { loggerFor } from '../../logger';
+const L = loggerFor(__filename);
 
 export const validateRegistrationData = async (
```

Comment:
A lot of changes here to switch from using alot of db queries to use an in memory cached indexes of all program donors